### PR TITLE
Conformance Behaviors Mapping for sig-apps

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/area conformance
/assign @johnbelamaric 
/assign @spiffxp

### CODE REVIEW

File: test/conformance/behaviors/sig-apps/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+suite: daemonset
+description: Base suite for daemonsets
+behaviors:
+- id: apps/daemonset/create
+  description: When a DaemonSet is created, a daemon pod must be launched on every node in the cluster.
```
[the remark this answers is not in the document]
Comment (spiffxp) [reply test/conformance/behaviors/sig-apps/daemonset.yaml:5]: s/must/MUST if we're still aiming to use RFC2119 keywords

**Review by spiffxp — COMMENTED** [review]
Some general questions, I haven't gone through the behavior descriptions exhaustively

File: test/conformance/behaviors/sig-apps/daemonset.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jefftree) [reply test/conformance/behaviors/sig-apps/daemonset.yaml:5]: Ahh wasn't aware of RFC 2119. I think a couple other places will need to be reworded as well. Thanks!

File: test/conformance/behaviors/sig-apps/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+suite: daemonset
```
[the remark this answers is not in the document]
Comment (Jefftree) [reply test/conformance/behaviors/sig-apps/daemonset.yaml]: Thanks for the catch. Updated the suite name for all the remaining yaml files.

File: test/conformance/behaviors/sig-apps/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+description: Base suite for daemonsets
+behaviors:
+- id: apps/daemonset/create
+  description: When a DaemonSet is created, a daemon pod must be launched on every node in the cluster.
+- id: apps/daemonset/create-node-selector
+  description: A DaemonSet's daemon pod should support node selection via label selectors. Daemon pods must only created on nodes matching the node selector.
```
[the remark this answers is not in the document]
Comment (Jefftree) [reply test/conformance/behaviors/sig-apps/daemonset.yaml]: Updated the language. At this early stage, I feel like creating a glossary will just spread out our information too much, adding yet another doc to read. I'm okay with just keeping "schedulable" here and diving deeper in sig-scheduling behaviors, but will defer to @johnbelamaric.

**Review by spiffxp — COMMENTED** [review]
feedback on daemonset

**Pull request closed without merging**